### DISCUSSION TITLE
OOM when running model `stt_en_conformer_ctc_large`

### DISCUSSION THREAD

**Discussion by eschmidbauer** [NVIDIA-NeMo/NeMo#2834] [Q&A] [answered]
Hi, 
im trying to test ASR with the stt_en_conformer_ctc_large model on an instance with 15G RAM, it works well for small files but when i try a 2.8mb file, the python script uses all 15G RAM and the kernel's OOM kills the process. I've read that the solution to larger files is to use VAD but I haven't been able to find any documentation or examples on how this can be done
Here is an example output where OOM kills the script:

![Screen Shot 2021-09-16 at 12 20 52 PM](https://user-images.githubusercontent.com/7139998/133648916-aab99f03-c7d7-458b-a8ec-1a4f0bc1e59b.png)
.

**Comment by titu1994** [comment, marked as answer]
Any audio above 20-30 seconds will oom with Conformer. You can use buffered audio evaluation with conformer to step around that issue 

https://github.com/NVIDIA/NeMo/blob/main/examples/asr/speech_to_text_buffered_infer.py

Note: change the modek stride to 4 instead of 8 for conformer and use larger chunk size  (upto 10-15 sec) for more accurate transcriptions. 

Tutorial describing this - https://github.com/NVIDIA/NeMo/blob/main/tutorials/asr/Streaming_ASR.ipynb

**Reply by eschmidbauer** [reply]
I will give this a try out and report an update- thank you!

**Reply by eschmidbauer** [reply]
I was able to put together a test script using the links you shared as a reference. thank you for the guidance!
